### ISSUE TITLE
Fatal error with v1.1.0 and PHP 7.2 / 7.3

### ISSUE THREAD

**Issue by axzx** [php-fig/container#30]
upgrade to version 1.1 broken many sites:(

---

Edit by @mnapoli 

Here is the exception message:

> PHP Fatal error:  Class Symfony\Component\DependencyInjection\Exception\ExceptionInterface cannot implement previously implemented interface Throwable in vendor/symfony/dependency-injection/Exception/ExceptionInterface.php on line 22

**Comment by tinogo** [comment]
After updating the `psr/container` package to version 1.1.0 I'm getting this error too.

Fixed it by extending the `require` section of my composer.json with...

````php
    "psr/container": "~1.0.0",
````

...and running `composer update` (resp. `composer update psr/container`) afterwards.

(or use the solution from @mynameisbogdan, which should be even better suited for this).

**Comment by mynameisbogdan** [comment]
We'll have to add a conflict rule to `composer.json`

```json
{
    "conflict": {
        "psr/container": "1.1"
    }
}
```

**Comment by akinspe** [comment]
This totally breaks symphony. I don't understand why this was added as a 1.1.0. It should have been 2.0

**Comment by mnapoli** [comment]
As @Ocramius pointed, this bug is fixed in PHP 7.4:

https://github.com/php-fig/container/pull/20#discussion_r588445994

https://3v4l.org/vUsLI

That means that v1.1.0 is currently:

- broken for PHP 7.2 / 7.3
- OK for PHP 7.4+

1: We probably need a 1.1.1 that solves the problem for PHP 7.2 & 7.3, we should probably revert the change and tag.

2: Then we have the possibility to re-introduce the change + up the PHP version requirement to PHP 7.4.

Does `2` make sense?

**Comment by moufmouf** [comment]
@mnapoli :+1:

(2) makes a lot of sense (but we can test it quietly later... not a Friday night :D )

**Comment by axzx** [comment]
> @mnapoli +1
> 
> (2) makes a lot of sense (but we can test it quietly later... not a Friday night :D )

Friday after 4 p.m. is the best time to put the code on production :)

**Comment by derrabus** [comment]
> 2: Then we have the possibility to re-introduce the change + up the PHP version requirement to PHP 7.4.
> 
> Does `2` make sense?

I don't think this is necessary. The change would be reintroduced with 2.0.0 anyway.

**Comment by Ocramius** [comment]
No real need for `2.0.0`:

 * `1.1.1` reverts the addition of `extends Throwable`
 * `1.2.0` adds `^7.4` to constraints, then re-adds `extends Throwable`
 
 That should be making the change smoother.

**Comment by derrabus** [comment]
> No real need for `2.0.0`:

A 2.0.0 has already been tagged, with added return type declarations.

**Comment by moufmouf** [comment]
And 2.0.0 is targetting PHP 7.2+ so is potentially having the same issue.

See #32 regarding a discussion on what could be done regarding 2.0.0

**Comment by derrabus** [comment]
Thank you for reacting so quickly and releasing version 1.1.1! ❤️

**Comment by Jean85** [comment]
Thank you for the really fast fix!

I'll take care of highlightin this pitfall in the evolution bylaw to avoid repeating the same error in the future.

Also, I agree with the plan for 1.1.2, totally makes sense.

**Comment by Jean85** [comment]
This is fixed for 1.x in 1.1.1.

I'll leave this open until it's fixed in 2.0.
